README.md

```diff
@@ -22,28 +32,147 @@ The Vault library is a required dependency of the [EOSIO SDK for Swift: Vault Si
+
+Generally, signature providers are called by [`EosioTransaction`](https://github.com/EOSIO/eosio-swift/blob/master/Sources/EosioSwift/EosioTransaction/EosioTransaction.swift) during signing. ([See an example here.](https://github.com/EOSIO/eosio-swift#basic-usage)) If you find, however, that you need to get available keys or request signing directly, this library can be invoked as follows:
+
+```swift
+let signProvider = try? EosioVaultSignatureProvider(accessGroup: "YOUR_ACCESS_GROUP")
+let publicKeysArray = signProvider?.getAvailableKeys() // Returns the public keys.
+```
+
+_[Learn more about Access Groups here.](https://developer.apple.com/documentation/security/keychain_services/keychain_items/sharing_access_to_keychain_items_among_a_collection_of_apps)_
+
+To sign an [`EosioTransaction`](https://github.com/EOSIO/eosio-swift/blob/master/Sources/EosioSwift/EosioTransaction/EosioTransaction.swift), create an [`EosioTransactionSignatureRequest`](https://github.com/EOSIO/eosio-swift/blob/master/Sources/EosioSwift/EosioSignatureProviderProtocol/EosioSignatureProviderProtocol.swift) object and call the `EosioVaultSignatureProvider.signTransaction(request:completion:)` method with the request:
+
+```swift
+var signRequest = EosioTransactionSignatureRequest()
+signRequest.serializedTransaction = serializedTransaction
+signRequest.publicKeys = publicKeys
+signRequest.chainId = chainId
+
+signProvider.signTransaction(request: signRequest) { (response) in
+    ...
+}
+```
+
+## Vault Signature Provider Library Methods
+
+This library is an implementation of [`EosioSignatureProviderProtocol`](https://github.com/EOSIO/eosio-swift/blob/master/Sources/EosioSwift/EosioSignatureProviderProtocol/EosioSignatureProviderProtocol.swift). It implements the following protocol methods:
+
+* `EosioVaultSignatureProvider.signTransaction(request:completion:)` signs an [`EosioTransaction`](https://github.com/EOSIO/eosio-swift/blob/master/Sources/EosioSwift/EosioTransaction/EosioTransaction.swift).
+* `EosioVaultSignatureProvider.getAvailableKeys(...)` returns an array containing the public keys associated with the private keys that the object is initialized with.
+
+To initialize the implementation:
+
+* `EosioVaultSignatureProvider.init(accessGroup:requireBio:)` initializes the signature provider.
+  * `accessGroup`: [Learn more about Access Groups here.](https://developer.apple.com/documentation/security/keychain_services/keychain_items/sharing_access_to_keychain_items_among_a_collection_of_apps)
+  * `requireBio`: Defaults to `false`. Some keys may require biometric authentication no matter what this flag is set to. For keys that do not require biometric authentication themselves, this flag can force the bio check.
+
+Other Keychain and/or Secure Enclave functionality can be accessed by calling methods directly on [EOSIO SDK for Swift: Vault](https://github.com/EOSIO/eosio-swift-vault), which is included with this library as a dependency.
```

Comment:
This line might need massaging. It appears to still be written assuming vault is in a different repo.

Reply:
Oops!  Yeah missed that one.  Updated to:

Other Keychain and/or Secure Enclave functionality can be accessed by calling methods directly on the Vault component, which is included with Vault Signature Provider as a dependency.